### PR TITLE
Add HC-256 cipher implementation

### PR DESCRIPTION
**Pull request by emc2**
This PR adds an implementation of the HC-256 cipher.  HC-256 was introduced in 2004 by Hongjun Wu, and is one of the finalists in the eSTREAM cipher competition.

### CODE REVIEW

**Comment by newpavlov** [comment]
Can you please relicense your code under dual MIT/Apache-2.0 license? Also currently crates support Rust versions up to 1.27, so also please use 2015 edition.

**Comment by emc2** [comment]
I'm currently overloaded, but I grant permission to relicense under both those licenses.  I will circle back and deal with the other requests eventually, but if you want to pick up the patch sooner, that's ok.

File: hc-256/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    /// Key size in bytes
+    type KeySize = U32;
+    /// Nonce size in bytes
+    type NonceSize = U32;
+
+    fn new(key: &GenericArray<u8, Self::KeySize>,
+           iv: &GenericArray<u8, Self::NonceSize>) -> Self {
+        let mut out = HC256::create();
+
+        out.init(key.as_slice(), iv.as_slice());
```
Comment (emc2) [inline hc-256/src/lib.rs]: Need to make sure this is decoding u8 arrays as little-endian

**Comment by emc2** [comment]
Note: I've addressed the original requests

**Comment by emc2** [comment]
Realized I'd forgotten to update this, and also the Zeroize version had gone stale.

**Pull request merged**